youtube_dl/extractor/cbs.py

```diff
@@ -20,6 +24,101 @@ def _parse_smil_subtitles(self, smil, namespace=None, subtitles_lang='en'):
+                ep['episode_title']))
+
+        return self.playlist_result(entries, playlist_title=carousel_title)
+
+    def _real_extract(self, url):
+        show_name = self._match_id(url)
+        webpage = self._download_webpage(url, show_name)
+
+        # not-quite JSON, no double-quotes:
+        #  var show = new CBS.Show({id:61456254});
+        show_id_json = self._search_regex(r'new CBS\.Show\(([^)]*)\);', webpage, 'show_id')
+
+        show = self._parse_json(show_id_json, show_name, transform_source=js_to_json)
+
+        # Found in http://www.cbs.com/assets/min/js/min/com.cbs.min.js?20170303-224247
+        # unminified at http://www.cbs.com/assets/js/min/com.cbs.js
+        # http://www.cbs.com/carousels/shows/61456254/offset/0/limit/15/xs/0/
+        # => {id: 240172, title: "Full Episodes",
+        episodes_url = urljoin(url, '/carousels/shows/%d/offset/0/limit/15/xs/0/' % show['id'])
+
+        #  var loader = new CBS.V2.CarouselLoader({
+        #                        'video-preview-carousel': function(element) {
+        #                            element.videoCarousel({
+        #                                id          : 241426,
+        #                                templates   : 'carousels/videoAdaptive',
+        #                                scroll      : 3,
+        #                                layout      : 3,
+        #                                start       : 0,
+        #                                saveState   : false
+        #                            });
+        #                        }
+        try:
+            clipdata = self._parse_json(
+                self._search_regex(r'element\.videoCarousel\(([^)]*)\);', webpage,
+                                   'clip carousel'),
+                show_name, transform_source=js_to_json)
+
+            # http://www.cbs.com/carousels/videosBySection/241426/offset/0/limit/15/xs/0/
+            # => {id: 241426, title: "Clips",
+            clips_url = urljoin(url,
+                    '/carousels/videosBySection/%d/offset/0/limit/15/xs/0' % clipdata['id'])
+            clips = self.carousel_playlist(clips_url, 'clips')
+        except RegexNotFoundError:
+            clips = { 'entries': [] }
+
+        # We separately retrieve a carousel of full episodes, and also one of clips.
+        # Clips are identifiable as such because they lack an "episode_number" field,
+        # unlike full episodes.
+        #
+        # It might be desirable to specify only retrieving a playlist of one or the other,
+        # but there isn't a good way for users to pass such parameters to InfoExtractors
+        # (custom URLs, maybe? With cbs: URLs?).
+        #
+        # But since the playlist is filterable, only full episodes can be returned with:
+        #   youtube_dl --match-filter 'episode_number' http://...
+        # and similarly, only clips can be returned with:
+        #   youtube_dl --match-filter '!episode_number' http://...
```

Comment:
If both are present it makes sense to extract both as well. This comment may be useful for end users but not for developers and it steels attention and takes time to read. So I vote for removing it from the source code.